doc/neps/nep-0054-simd-cpp-highway.rst

```diff
@@ -0,0 +1,368 @@
+written in its README, *This is not an officially supported Google product*).
+That is not a bad thing; it means that it is less likely to go unsupported due
+to a Google business decision about the project. Quite a few well-known open
+source projects under the ``google`` GitHub org state this, e.g. `JAX`_ and
+`tcmalloc`_.
+
+
+Supported features/targets
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Both frameworks support a large set of platforms and SIMD instruction sets,
+as well as generic scalar/fallback versions. The main differences right now are:
+
+- NumPy supports IBM Z-system (s390x, VX/VXE/VXE2) while Highway supports Z14, Z15.
+- Highway supports ARM SVE/SVE2 and RISC-V RVV (sizeless instructions), while
+  NumPy does not.
+
+  - The groundwork for sizeless SIMD support in NumPy has been done in
+    `gh-21057`_, however SVE/SVE2 and RISC-V are not yet implemented there.
+
+There is also a difference in the granularity of instruction set groups: NumPy
+supports a more granular set of architectures than Highway. See the list of
+targets for Highway `here <https://github.com/google/highway/#targets>`__
+(it's roughly per CPU family) and for NumPy
+`here <https://numpy.org/doc/1.25/reference/simd/build-options.html#supported-features>`__
+(roughly per SIMD instruction set). Hence with Highway we'd lose some
+granularity - but that is probably fine, we don't really need this level of
+granularity, and there isn't much evidence that users explicitly play with this
+to squeeze out the last bit of performance for their own CPU.
+
+
+Compilation strategy for multiple targets and runtime dispatching
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Highway compiles once while using preprocessing tricks to generate multiple
+stanzas for each CPU feature within the same compilation unit (see the
+``foreach_target.h`` usage and dynamic dispatch docs for how that is done).
+Universal Intrinsics generate multiple compilation units, one for each CPU
+feature group, and compiles multiple times, linking them all together (with
+different names) for runtime dispatch. The Highway technique may not work
+reliably on MSVC, the Universal Intrinsic technique does work on MSVC.
+
+Which one is more robust? The experts disagree. Jan thinks that the Highway
+approach is more robust and in particular avoids the linker pulling in
+functions with too-new instructions into the final binary. Sayed thinks that
+the current NumPy approach (also used by OpenCV) is more robust, and in
+particular is less likely to run into compiler-specific bugs or catch them
+earlier. Both agree the meson build system allows specifying object link order,
+which produces more consistent builds. However that does tie NumPy to meson.
+
+Matti thinks the current build strategy is working well for NumPy and the
+advantages of changing the build and runtime dispatch, with possible unknown
+instabilities outweighs the advantages that fully adopting Highway may bring.
```

Comment:
```suggestion
instabilities outweighs the advantages that adopting Highway's dynamic dispatch may bring.
```

Reply:
I incorporated these comments in the last commit